Source/Scene/Cesium3DTileOptimizations.js

```diff
@@ -0,0 +1,122 @@
+/*global define*/
+define([
+        '../Core/Cartesian3',
+        '../Core/defaultValue',
+        '../Core/freezeObject',
+        './Cesium3DTileOptimizationHint',
+        './TileOrientedBoundingBox',
+        './TileBoundingRegion'
+    ], function(
+        Cartesian3,
+        defaultValue,
+        freezeObject,
+        Cesium3DTileOptimizationHint,
+        TileOrientedBoundingBox,
+        TileBoundingRegion) {
+    'use strict';
+
+    /**
+     * Defines optional optimization hints for a {@link Cesium3DTileset}.
+     * Optimizations marked as {@link Cesium3DTileOptimizationHint.USE_OPTIMIZATION} or {@link Cesium3DTileOptimizationHint.SKIP_OPTIMIZATION}
+     * will not be evaluated at runtime
+     *
+     * @private
+     * 
+     * @alias Cesium3DTileOptimizations
+     * @constructor
+     */
+    function Cesium3DTileOptimizations(options) {
```

Comment:
To avoid some indirection at runtime (and likely cache misses), consider replacing with with a boolean property on the tile.  So file would just contain a single function for `checkChildrenWithinParent`.

In time, the boolean would probably become a bitmask with many booleans for different optimizations and tiles would more cohesively fit into cache lines (I'm not saying that they do today, but this extra indirection does not help the case).